include/nbl/video/CThreadSafeGPUQueueAdapter.h

```diff
@@ -4,30 +4,21 @@
 
 namespace nbl::video
 {
+    /*
+        A thread-safe implementation of IGPUQueue.
+        Note, that using the same queue as both a threadsafe queue and a normal queue invalidates the safety.
+    */
     class CThreadSafeGPUQueueAdapter : public IGPUQueue
     {
     protected:
-        nbl::core::smart_refctd_ptr<IGPUQueue> originalQueue = nullptr;
+        core::smart_refctd_ptr<IGPUQueue> originalQueue = nullptr;
         std::mutex m;
     public:
-        CThreadSafeGPUQueueAdapter(nbl::core::smart_refctd_ptr<IGPUQueue>&& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()), originalQueue(originalQueue) {}
-        CThreadSafeGPUQueueAdapter(CThreadSafeGPUQueueAdapter&& other) : IGPUQueue(nullptr, other.originalQueue->getFamilyIndex(), other.originalQueue->getFlags(), other.originalQueue->getPriority()), originalQueue(other.originalQueue) {}
-        
-        CThreadSafeGPUQueueAdapter(const CThreadSafeGPUQueueAdapter& other) : IGPUQueue(nullptr, other.originalQueue->getFamilyIndex(), other.originalQueue->getFlags(), other.originalQueue->getPriority()), originalQueue(other.originalQueue) {}
-        CThreadSafeGPUQueueAdapter(const nbl::core::smart_refctd_ptr<IGPUQueue>& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()), originalQueue(originalQueue) {}
-        
+        CThreadSafeGPUQueueAdapter(nbl::core::smart_refctd_ptr<IGPUQueue>&& originalQueue) : IGPUQueue(nullptr, originalQueue->getFamilyIndex(), originalQueue->getFlags(), originalQueue->getPriority()),
+            originalQueue(originalQueue) {}        
```

Comment:
`std::move(originalQueue)`

also you have shadowing variables, you cannot call a member and argument the same thing (well you can, but you'll just confuse the crap out of everyone, and its not something you want to reason about and test the compiler parser with)

Reply:
ok, but what's the point of passing an rvalue to `std::move`?